lockfile/src/parsers/yarn.rs

```diff
@@ -29,28 +35,80 @@ fn entry(input: &str) -> IResult<&str, Package> {
     Ok((i, my_entry))
 }
 
-fn parse_entry(input: &str) -> IResult<&str, Package> {
-    context("entry", tuple((entry_name, entry_version)))(input).map(|(next_input, res)| {
-        let (name, version) = res;
-        (next_input, Package {
-            name: name.to_string(),
-            version: PackageVersion::FirstParty(version.to_string()),
-            package_type: PackageType::Npm,
-        })
-    })
+fn parse_entry(input: &str) -> IResult<&str, Option<Package>> {
+    let (input, (name, version)) = context("entry", tuple((entry_name, entry_version)))(input)?;
+
+    let version = match version {
+        Some(version) => version,
+        None => return Ok((input, None)),
+    };
```

Comment:
nit: seems like a good place for let/else.
```suggestion
    let Some(version) = version else {
        return Ok((input, None));
    };
```

Reply:
I do not think `let else` is ever a better option.

Reply:
Interesting, why's that?

Reply:
It's pointless extra syntax that doesn't make code more readable. There's no reason to add new syntax for readers just to save a single line. It's addition to Rust is completely misguided imo. Makes a language that already has way too much special syntax even harder to learn and read.

Reply:
Fair enough 👍